### PR TITLE
playground: Add left panel and use brand colors

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR uses Astral's brand color in the playground more consistently and adds a left panel to switch between file and settings. 

## Test Plan

[Screencast from 2023-07-17 19-35-11.webm](https://github.com/astral-sh/ruff/assets/1203881/0fc241b6-2607-4451-887e-23cd613a0187)

I promise it shows an error when the settings file contains a syntax errors. It just happens that it wasn't part of the selection that I recorded.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5830** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5830" target="_blank">Graphite</a> 
    * **PR #5831** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5831" target="_blank">Graphite</a> 
      * **PR #5838** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5838" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5838?utm_source=stack-comment).

**Comment by MichaReiser** [comment]
@charliermarsh feel free to make changes if you don't like specific details. This is just not-designer Micha who tried to follow our brand guideline

File: playground/src/Editor/SideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      role="button"
+      className={`py-4 px-2 relative ${
+        selected ? "fill-white" : "fill-slate-500"
+      }`}
+    >
+      {children}
+      {selected && (
+        <span className="absolute start-0 inset-y-0 bg-white w-0.5 rounded-full transition-opacity duration-150 opacity-100"></span>
+      )}
+    </li>
```
Comment (charliermarsh) [inline playground/src/Editor/SideBar.tsx:59]: Are the buttons horizontally centered in the sidebar? I can't tell from the video.

**Review by charliermarsh — APPROVED** [review]
LGTM! I might do a follow-up pass when I have time on some of the brand-specific stuff like the button styling, but this is great.

File: playground/src/Editor/AstralButton.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "dark:hover:outline-radiate",
+        "rounded-md",
+        "text-sm",
+        "font-semibold",
+        "enabled:hover:bg-galaxy",
+        className,
+      )}
+      {...otherProps}
+    >
+      {children}
```
Comment (charliermarsh) [inline playground/src/Editor/AstralButton.tsx:27]: Maybe we should use small caps for this, like the button on the top right of the homepage: https://astral.sh/. Maybe the same font too? Wdyt?

File: playground/src/Editor/Header.tsx [hunk 3]
```diff
@@ -29,77 +25,58 @@ export default function Header({
...
-        <div className="hidden sm:block mx-6 lg:mx-4 w-px h-6 bg-gray-200 dark:bg-gray-700" />
+        <Divider />
         <RepoButton />
-        <div className="hidden sm:block mx-6 lg:mx-4 w-px h-6 bg-gray-200 dark:bg-gray-700" />
+        <Divider />
         <div className="hidden sm:block">
           <ShareButton key={edit} onShare={onShare} />
         </div>
-        <div className="hidden sm:block mx-6 lg:mx-4 w-px h-6 bg-gray-200 dark:bg-gray-700" />
+        <Divider />
```
Comment (charliermarsh) [inline playground/src/Editor/Header.tsx:71]: Are the dividers different colors? I assume it's just an artifact of the recording or something.

Screen Shot 2023-07-18 at 8 49 18 PM

File: playground/src/Editor/SideBar.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { FileIcon, SettingsIcon } from "./Icons";
+import { type ReactNode } from "react";
+
+type Tool = "Settings" | "Source";
+
+type SideBarProps = {
+  selected: Tool;
+  onSelectTool(tool: Tool): void;
+};
```
Comment (charliermarsh) [inline playground/src/Editor/SideBar.tsx:9]: I always just inline single-use definitions like this, e.g.:

```typescript
export default function SideBar({ selected, onSelectTool }: { selected: Tool, ... } { ... }
```

But it's entirely subjective :)

**Comment by charliermarsh** [comment]
I'm triggered by the juxtaposition of the dark purple (header and sidebar) against the black (editor) but I can take on this burden:

Screen Shot 2023-07-18 at 8 50 56 PM

File: playground/src/Editor/AstralButton.tsx (hunk 2 above)
Comment (MichaReiser) [reply playground/src/Editor/AstralButton.tsx:32]: I copied over some styles from astral. Is it okay to commit the fonts?

File: playground/src/Editor/SideBar.tsx (hunk 4 above)
Comment (MichaReiser) [reply playground/src/Editor/SideBar.tsx:9]: I can't do that. :stuck_out_tongue: 

I find it extremely hard to read, always mess up the syntax when writing the types, and you need to refactor them out once you start adding new props.

**Comment by MichaReiser** [comment]
> juxtaposition

I'm not sure if I made it worse :sweat_smile: 

My idea was to use `space` for the header and side bar. It looks kind of good for the side bar, but we're not supposed to use our logo on space. I now use `galaxy` for the code and header, but `space` for the sidebar. Let me know what you think


**Black code**
image

**Galaxy code**

image

```patch
Subject: [PATCH] Add right panel
---
Index: playground/src/Editor/MonacoThemes.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/playground/src/Editor/MonacoThemes.tsx b/playground/src/Editor/MonacoThemes.tsx
--- a/playground/src/Editor/MonacoThemes.tsx	(revision 6e1b1c0f364b369f0aebe511a0d70a87e5199f66)
+++ b/playground/src/Editor/MonacoThemes.tsx	(date 1689751756944)
@@ -389,7 +389,7 @@
         "tab.inactiveForeground": "#565b66",
         "tab.unfocusedActiveForeground": "#565b66",
         "tab.unfocusedInactiveForeground": "#565b66",
-        "editor.background": "#0b0e14",
+        "editor.background": "#261230",
         "editor.foreground": "#bfbdb6",
         "editorLineNumber.foreground": "#6c738099",
         "editorLineNumber.activeForeground": "#6c7380e6",
@@ -408,7 +408,7 @@
         "editor.findMatchHighlightBorder": "#5f4c7266",
         "editor.findRangeHighlightBackground": "#6c598040",
         "editor.rangeHighlightBackground": "#6c598033",
-        "editor.lineHighlightBackground": "#131721",
+        "editor.lineHighlightBackground": "#30173D",
         "editorLink.activeForeground": "#e6b450",
         "editorWhitespace.foreground": "#6c738099",
         "editorIndentGuide.background": "#6c738033",
Index: playground/src/Editor/SideBar.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/playground/src/Editor/SideBar.tsx b/playground/src/Editor/SideBar.tsx
--- a/playground/src/Editor/SideBar.tsx	(revision 6e1b1c0f364b369f0aebe511a0d70a87e5199f66)
+++ b/playground/src/Editor/SideBar.tsx	(date 1689751464419)
@@ -10,7 +10,7 @@
 
 export default function SideBar({ selected, onSelectTool }: SideBarProps) {
   return (
-    <ul className="w-12 flex-initial border-r bg-galaxy flex flex-col items-stretch">
+    <ul className="w-12 flex-initial border-r bg-space flex flex-col items-stretch">
       <SideBarEntry
         title="Source"
         onClick={() => onSelectTool("Source")}

```

File: playground/src/Editor/Header.tsx (hunk 3 above)
Comment (MichaReiser) [reply playground/src/Editor/Header.tsx:71]: I don't think so... They all render the same HTML 
image

**Comment by charliermarsh** [comment]
Let's run with what you had initially, I need to think on it a bit more and maybe play around with it but I def don't want to block shipping.

**Comment by MichaReiser** [comment]
> Let's run with what you had initially, I need to think on it a bit more and maybe play around with it but I def don't want to block shipping.

Feel free to delegate the changes once you know how you want them to be. I enjoyed this a lot (feel free to do I yourself, if you find it fun too)

**Pull request merged**